cueball/src/connection_pool.rs

```diff
@@ -645,6 +638,8 @@ where
             Some(conn) => {
                 if conn.has_broken() {
                     warn!(self.log, "Found an invalid connection, not returning to the pool");
+                    connection_data.stats.total_connections -= 1.into();
+                    connection_data.stats.idle_connections -= 1.into();
```

Comment:
I don't think we want to decrement this here since it represents the number of unused connections still in the pool and the broken claimed connection is not represented by that count.

Reply:
Good catch.